ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java

```diff
@@ -77,6 +77,9 @@ public void reportStageEnd(String stageName, Status status, long lastReplId) thr
       Stage stage = progress.getStageByName(stageName);
       stage.setStatus(status);
       stage.setEndTime(System.currentTimeMillis());
+      if (Status.FAILED == status) {
```

Comment:
Could you please give background of this?

Reply:
If the stage has failed, the overall state should also be failed. Now it remains in progress. Noticed during testing this issue.

Reply:
Sure. Can we have a test for this?

Reply:
Added